src/client.rs

```diff
@@ -966,28 +961,62 @@ impl Client {
         ))
     }
 
+    /// Mounts a blob to the provided reference, from the given source
+    pub async fn mount_blob(
+        &self,
+        image: &Reference,
+        source: &Reference,
+        digest: &str,
+    ) -> Result<()> {
+        let base_url = self.to_v2_blob_upload_url(image);
+        let url = format!("{}?mount={}&from={}", base_url, digest, source.repository());
+
+        let res = RequestBuilderWrapper::from_client(self, |client| client.post(url.clone()))
+            .apply_auth(image, RegistryOperation::Push)?
+            .into_request_builder()
+            .send()
+            .await?;
+
+        self.extract_location_header(image, res, &reqwest::StatusCode::CREATED)
+            .await?;
+
+        Ok(())
+    }
+
     /// Pushes the manifest for a specified image
     ///
     /// Returns pullable manifest URL
     async fn push_manifest(&self, image: &Reference, manifest: &OciManifest) -> Result<String> {
-        let url = self.to_v2_manifest_url(image);
-
-        let mut headers = HeaderMap::new();
-        let content_type = manifest.content_type();
-        headers.insert("Content-Type", content_type.parse().unwrap());
-
         // Serialize the manifest with a canonical json formatter, as described at
         // https://github.com/opencontainers/image-spec/blob/main/considerations.md#json
         let mut body = Vec::new();
         let mut ser = serde_json::Serializer::with_formatter(&mut body, CanonicalFormatter::new());
         manifest.serialize(&mut ser).unwrap();
 
+        self.push_manifest_raw(image, body, manifest.content_type().parse().unwrap())
+            .await
+    }
+
+    /// Pushes the manifest, provided as raw bytes, for a specified image
+    ///
+    /// Returns pullable manifest url
+    pub async fn push_manifest_raw(
+        &self,
+        image: &Reference,
+        body: Vec<u8>,
+        content_type: HeaderValue,
```

Comment:
This might be a cleaner API if this were to be `H` where `H: TryInto<HeaderValue, Error = OciDistributionError>` so people can pass anything that can be turned into a header value. Any thoughts on how you think this will be used?

Reply:
I am using this function in an application that creates an OCI image in memory and pushes the result to the registry (see [this blog post](https://ochagavia.nl/blog/crafting-container-images-without-dockerfiles/) for details). The specific code is:

```rust
registry_client
    .push_manifest_raw(
        new_image_reference,
        image_manifest_json,
        OCI_IMAGE_MEDIA_TYPE.parse().unwrap(),
    )
    .await?;
```

I experimented changing the line you mentioned to `content_type: impl TryInto<HeaderValue, Error = OciDistributionError>` and ran into problems, but `impl TryInto<HeaderValue, Error = InvalidHeaderValue>` did the trick ;)